libs/core/langchain_core/runnables/base.py

```diff
@@ -4529,14 +4531,18 @@ def __eq__(self, other: Any) -> bool:
 
     def __repr__(self) -> str:
         """A string representation of this Runnable."""
-        if hasattr(self, "func") and isinstance(self.func, itemgetter):
-            return f"RunnableLambda({str(self.func)[len('operator.'):]})"
-        elif hasattr(self, "func"):
-            return f"RunnableLambda({get_lambda_source(self.func) or '...'})"
-        elif hasattr(self, "afunc"):
-            return f"RunnableLambda(afunc={get_lambda_source(self.afunc) or '...'})"
-        else:
-            return "RunnableLambda(...)"
+        if self._repr is None:
```

Comment:
similar comment to last one - why not decorate with `@functools.lru_cache` instead?

Reply:
Using `lru_cache` on instance methods can lead to memory leaks (as `self` is hard referenced by the cache)
See https://docs.astral.sh/ruff/rules/cached-instance-method/
Do you prefer the solution with an external function or the attribute memoizer ?

Reply:
TIL